File: Master 1/Python && R/Exos/plus_ou_moins.py
import random

def generate_number():
    return random.randint(1, 100)

def guess_my_number():
    print("\n\n######################################################################### \n\n")
    print("Bienvenue au jeu du plus ou moins !")
    print("Je choisi un nombre entre 1 et 100.")
    print("Essayez de le deviner en choisissnat des numeros ou aller à la fin avec la command 'fin'.")

    print("\n\n######################################################################### \n\n")
    
    secret_number = generate_number()
    attempts = 0

    while True:
        print("\n\n###############\n\n")
        user_input = input("Entrez votre choix: ")

        if user_input.lower() == 'fin':
            print("\n\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n")
            print(f"Vous ne savez plus quoi choisir ! D' accord, le nombre était {secret_number}")
            print("\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n\n")
            break

        # Validate input
        if not user_input.isdigit():
            print("Veuillez entrer un nombre entier.")
            continue

        estimate = int(user_input)

        attempts += 1

        if estimate > secret_number:
            print("entrez un nombre plus petit! ")
            print("\n\n###############\n")
        elif estimate < secret_number:
            print("Entrez un nombre plus grand !")
            print("\n\n###############\n")
        else:
            print("\n\n***********************************************************\n")
            print(f"Félicitations ! Vous avez trouvé le nombre en {attempts} tentatives.")
            print("\n\n***********************************************************\n")
            break

if __name__ == "__main__":
    guess_my_number()

File: Master 1/Python && R/Exos/plus_ou_moins.R
# Fonction pour générer un nombre aléatoire entre 1 et 100
generate_number <- function() {
  return(sample(1:100, 1))
}

# Fonction principale du jeu
guess_my_number <- function() {
  # Afficher les instructions initiales
  cat("\n\n######################################################################### \n\n")
  cat("Bienvenue au jeu du plus ou moins !\n")
  cat("Je choisis un nombre entre 1 et 100.\n")
  cat("Essayez de le deviner en choisissant des nombres ou allez à la fin avec la commande 'fin'.\n")
  cat("\n\n######################################################################### \n\n")
  
  # Générer le nombre secret
  secret_number <- generate_number()
  attempts <- 0

  # Boucle principale du jeu
  while (TRUE) {
    cat("\n\n###############\n")
    
    # Obtenir l'entrée de l'utilisateur
    user_input <- readline("Entrez votre choix: ")
    
    # Vérifier si l'utilisateur souhaite quitter
    if (tolower(user_input) == 'fin') {
      cat("\n\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n")
      cat(paste("Vous ne savez plus quoi choisir ! D'accord, le nombre était", secret_number))
      cat("\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n\n")
      break
    }
    
    # Valider l'entrée
    if (!grepl("^[0-9]+$", user_input)) {
      cat("Veuillez entrer un nombre entier.\n")
      next
    }
    
    # Convertir l'entrée en entier
    estimate <- as.integer(user_input)
    
    # Incrémenter le compteur de tentatives
    attempts <- attempts + 1
    
    # Comparer avec le nombre secret
    if (estimate > secret_number) {
      cat("Entrez un nombre plus petit!\n")
      cat("\n\n###############\n")
    } else if (estimate < secret_number) {
      cat("Entrez un nombre plus grand!\n")
      cat("\n\n###############\n")
    } else {
      cat("\n\n***********************************************************\n")
      cat(paste("Félicitations ! Vous avez trouvé le nombre en", attempts, "tentatives."))
      cat("\n\n***********************************************************\n")
      break
    }
  }
}

# Lancer le jeu
guess_my_number()
